pyirf/interpolation/visible_edges_extrapolator.py

```diff
@@ -0,0 +1,213 @@
+    target_point: numpy.ndarray, shape=(1, M)
+        Value for which the extrapolation is performed (target point)
+
+    Returns
+    -------
+    extrapolation_weights: np.ndarray, shape=(L)
+        Weights for each visible facet, corresponding to the extrapolation
+        weight for the respective triangulation simplex. Weigths sum to unity.
+
+    References
+    ----------
+    .. [1] P. Alfred (1984). Triangular Extrapolation. Technical summary rept.,
+    Univ. of Wisconsin-Madison. https://apps.dtic.mil/sti/pdfs/ADA144660.pdf
+    """
+
+    angles = np.array(
+        [
+            point_facet_angle(line, target_point.squeeze())
+            for line in visible_facet_points
+        ]
+    )
+    weights = np.arccos(angles) ** (m + 1)
+
+    return weights / weights.sum()
+
+
+class ParametrizedVisibleEdgesExtrapolator(ParametrizedNearestSimplexExtrapolator):
+    """
+    Extrapolator class extending the ParametrizedNearestSimplexExtrapolator.
+
+    While the ParametrizedNearestSimplexExtrapolator does not result in a smooth
+    extrapolation outside of the grid due to using only the nearest available
+    simplex, this extrapolator blends over all visible edges as discussed in [1].
+    For one grid-dimension this is equal to the ParametrizedNearestSimplexExtrapolator,
+    the same holds for grids consisting of only one simplex or constellations,
+    where only one simplex is visible from a target.
+
+    Parameters
+    ----------
+    grid_points: np.ndarray, shape=(N, ...)
+        Grid points at which templates exist. May be one ot two dimensional.
+        Have to be sorted in accending order for 1D.
+    params: np.ndarray, shape=(N, ...)
+        Array of corresponding parameter values at each point in grid_points.
+        First dimesion has to correspond to number of grid_points
+    m: non-zero int >= 1
+        Degree of smoothness wanted in the extrapolation region. See [1] for
+        additional information. Defaults to 1.
+
+    Raises
+    ------
+    TypeError:
+        If m is not a number
+    ValueError:
+        If m is not a non-zero integer
+
+    Note
+    ----
+        Also calls pyirf.interpolation.ParametrizedNearestSimplexExtrapolator.__init__.
+
+    References
+    ----------
+    .. [1] P. Alfred (1984). Triangular Extrapolation. Technical summary rept.,
+           Univ. of Wisconsin-Madison. https://apps.dtic.mil/sti/pdfs/ADA144660.pdf
+
+    """
+
+    def __init__(self, grid_points, params, m=1):
+        super().__init__(grid_points, params)
+
+        # Test wether m is a number
+        try:
+            m > 0
+        except TypeError:
+            raise TypeError(f"Only positive integers allowed for m, got {m}.")
+
+        # Test wether m is a finite, positive integer
+        if (m <= 0) or ~np.isfinite(m) or (m != int(m)):
+            raise ValueError(f"Only positive integers allowed for m, got {m}.")
+
+        self.m = m
+
+    def extrapolate(self, target_point):
```

Comment:
In sphinx, subclasses should automatically inherit docstrings of their baseclass.

So I think this should be left empty and the approach should be defined in the class docstring.